runtime/lua/vim/lsp.lua

```diff
@@ -355,6 +355,9 @@ end
 ---  - {resolved_capabilities} (table): Normalized table of
 ---    capabilities that we have detected based on the initialize
 ---    response from the server in `server_capabilities`.
+---
+---  - {messages} (table): Current progress information from $/progress and
```

Comment:
Just to be clear, a client can be shared across multiple buffers. And some of this implementation makes me think that the messages table isn't respecting that.

Reply:
I believe it respects that. A given `client.messages.status` is a map of URI (file/buffer) to status message.